marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/xquery-lib.xqy

```diff
@@ -64,5 +65,28 @@ declare function data-hub-map-to-canonical(
   return
     if ($format="xml")
     then inst:canonical-xml($results)
-    else inst:canonical-json($results)
+    else(
+      let $element := $results/element()
+      return
+      (: This change is necessitated by DHFPROD-6219. In case the entity doesn't have strucutred properties, it
+         returns empty json object. If it has structured properties, the empty structured properties are removed.
+         inst:canonical-json() method introduces the '$ref' .  :)
+      if (empty($element/*)) then
+        document{json:object()=>map:with(string(fn:node-name($element)), json:object())}
+      else
+        inst:canonical-json(xquery-lib:clean-xml($element))
+    )
+};
+
+
+declare function clean-xml($element as item()*) as document-node() {
```

Comment:
Let's call this something more descriptive than "clean", which doesn't describe what it's doing. I think your comment indicates what the function name should be - perhaps "remove-empty-structured-properties"?